examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,222 @@
+# iterations.
+#
+# Indeed, if the `alpha` parameters vary depending on the input data, it
+# might be challenging to put our model in production because we would not be
+# able to fix this hyperparameter.
+#
+# Let's check the `alpha` parameter variance.
+alpha = [est[-1].alpha_ for est in cv_estimators]
+plt.hist(alpha, bins=30, density=True)
+plt.xlabel("Alpha")
+plt.ylabel("Density")
+_ = plt.title("Distribution of alpha parameter \nduring cross-validation")
+
+# %%
+# We see that the regularization parameter, `alpha`, values are centered and
+# condensed around 40. This is a good sign and means that most of the models
+# tuned within the cross-validation had similar `alpha` values.
+#
+# However, not only hyperparameter such as `alpha` should be studied. The model
+# parameter coming out of the fitting process should analyzed. In our case, we
+# used a linear model. These models are parametrized and rely on two
+# parameters: `coef_` and `intercept_`. Therefore, we should analyze the
+# variance of these parameters as well.
+#
+# For the sake of simplicity, we are going to solely look at the `coef_`.
+import pandas as pd
+
+coefficients = pd.DataFrame([est[-1].coef_ for est in cv_estimators], columns=X.columns)
+coefficients
+
+# %%
+import seaborn as sns
+
+plt.figure(figsize=(9, 7))
+sns.stripplot(data=coefficients, orient="h", color="k", alpha=0.5)
+sns.boxplot(data=coefficients, orient="h")
+plt.axvline(x=0, color=".5")
+plt.title("Coefficient values our model")
+_ = plt.subplots_adjust(left=0.3)
+
+# %%
+# We observe that the coefficients vary minimally, meaning that all the
+# trained models are similar. Each individual model is expected to more or less
+# give the same predictions.
+#
+# Putting a predictive model in production
+# ----------------------------------------
+#
+# In the above analysis, we saw that the values of lambda did not vary much.
+# Indeed, we can retrain this model and optimized the value of `alpha` on the
+# full training set.
+from sklearn.base import clone
+
+production_model = clone(model)
+production_model.fit(X_train, y_train)
+
+# %%
+# At the beginning of the process, when we split our data, we had a set of left
+# out data. Now, we can use it to further check if the model performs as we
+# would expect from the analysis done within the cross-validation framework.
+print(
+    "The performance of our production model: "
+    f"R2={production_model.score(X_test, y_test):.2f}"
+)
+
+# %%
+# We see that the statistical performance is comparable to the cross-validation
+# statistical performances which is not surprising. Similarly, we could look at
+# the coefficients of the production model and compare them with the
+# coefficients obtained within the cross-validation study.
+#
+# However, you should be aware that this latest step does not give any
+# information about the variance of the model. It should never be used to
+# evaluate the model itself.
```

Comment:
What exaclty is meant by that?

Reply:
I think here I wanted again to point out that you get a single point and not a score distribution.

Reply:
I'm still confused. Let's say you evaluate MSE on the test set. Then, one contribution of your MSE sore comes from the variance of you model—variance of the model's prediction to be precise—the other from the bias term and the variance of the target.
If you mean variance of the algorithm, that's something slightly different. Additionally, the variance of the coefficients could be inferred in-sample (not very unbiased with the cross validated alpha, but still). But for some reasons, this is not done in scikit-learn.